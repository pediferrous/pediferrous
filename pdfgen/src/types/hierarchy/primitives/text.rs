//! Implementation of PDF Text object.

use super::{font::Font, object::Object, rectangle::Position, string::PdfString};

/// Comment
#[derive(Debug)]
pub struct TextTransform {
    /// Comment
    position: Position,

    /// Comment
    size: u32,
}

/// Comment
#[derive(Debug)]
pub struct Text {
    /// Comment
    content: PdfString,

    /// Comment
    transform: TextTransform,

    /// Comment
    font: Font,
}

impl Text {
    /// Comment
    fn new(content: &str) -> Self {
        Self { content: content, transform TextTransform::default(), font: Font::default() }
    }
}
