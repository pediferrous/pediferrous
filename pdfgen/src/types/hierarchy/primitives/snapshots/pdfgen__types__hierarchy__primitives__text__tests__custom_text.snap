---
source: pdfgen/src/types/hierarchy/primitives/text.rs
expression: output
snapshot_kind: text
---
BT
/CustomFnt 14 Tf
0 0 Td
(This is a custom text content.) Tj
ET
